src/components/v5/common/ActionSidebar/partials/motions/MotionSimplePayment/steps/FinalizeStep/FinalizeStep.tsx

```diff
@@ -1,10 +1,127 @@
-import React, { FC } from 'react';
+import React, { FC, useEffect, useState } from 'react';
+import { ActionTypes } from '~redux';
+
+import { formatText } from '~utils/intl';
+import Button, { PendingButton } from '~v5/shared/Button';
+import { ActionForm } from '~shared/Fields';
+import CardWithStatusText from '~v5/shared/CardWithStatusText';
+import DescriptionList from '../VotingStep/partials/DescriptionList';
+import { useClaimConfig, useFinalizeStep } from './hooks';
+import { FinalizeStepProps } from './types';
+// import TeamBadge from '~v5/common/Pills/TeamBadge';
+import { useAppContext, useColonyContext } from '~hooks';
+import Icon from '~shared/Icon';
+
+import { ActionButton } from '~shared/Button';
 
 const displayName =
   'v5.common.ActionSidebar.partials.motions.MotionSimplePayment.steps.FinalizeStep';
 
-const FinalizeStep: FC = () => {
-  return <div>FinalizeStep</div>;
+const FinalizeStep: FC<FinalizeStepProps> = ({
+  actionData,
+  startPollingAction,
+  stopPollingAction,
+  refetchAction,
+}) => {
+  const { user } = useAppContext();
+  const [isPolling, setIsPolling] = useState(false);
+  const { refetchColony } = useColonyContext();
+  const { isFinalizable, transform } = useFinalizeStep(actionData);
+  const {
+    items,
+    isClaimed,
+    buttonTextId,
+    remainingStakesNumber,
+    handleClaimSuccess,
+    claimPayload,
+    canClaimStakes,
+  } = useClaimConfig(actionData, startPollingAction, refetchAction);
+
+  const handleSuccess = () => {
+    startPollingAction(1000);
+    setIsPolling(true);
+  };
+
+  /* Stop polling when mounted / dismounted */
+  useEffect(() => {
+    stopPollingAction();
+    return stopPollingAction;
+  }, [stopPollingAction]);
+
+  /* Update colony object when motion gets finalized. */
+  if (actionData.motionData.isFinalized) {
+    refetchColony();
+  }
+
+  if (isClaimed) {
+    stopPollingAction();
+  }
```

Comment:
It's not valid to run side effects inside of a render body. Add a separate `useEffect` for this or move it to one of the existing ones

Reply:
fixed